src/Oowp/OowpRouterHelper.php

```diff
@@ -86,7 +86,15 @@ public function querySingle($args, $redirectCanonical = false) {
             $redirectCanonical = false;
         }
 
-        $query = $this->query($args);
+		$query = $this->query( $args );
+		//no matched posts, so first check if this is a logged in user with a private post
+		if ( ( ! count( $query ) ) && is_user_logged_in() ) {
+			$args['author']      = get_current_user_id();
+			$args['name']        = $args['name'];
+			$args['post_status'] = null;
```

Comment:
should this explicitly be ['private']? If not, could you add a comment to explain why